src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts

```diff
@@ -74,24 +74,19 @@ export class StudentExamDetailComponent implements OnInit {
     loadStudentExam() {
         this.courseId = Number(this.route.snapshot.paramMap.get('courseId'));
         this.examId = Number(this.route.snapshot.paramMap.get('examId'));
-        this.route.data.subscribe(({ studentExam }) => this.setStudentExam(studentExam));
-        this.calculateGrade();
+        this.route.data.subscribe(({ studentExam }) => this.setStudentExamWithGrade(studentExam));
     }
 
     /**
      * Sets grade related information if a grading scale exists for the exam
      */
-    calculateGrade() {
-        const achievedPercentageScore = (this.achievedTotalPoints / this.maxTotalPoints) * 100;
-        this.gradingSystemService.matchPercentageToGradeStepForExam(this.courseId, this.examId, achievedPercentageScore).subscribe((gradeObservable) => {
-            if (gradeObservable && gradeObservable.body) {
-                this.gradingScaleExists = true;
-                const gradeDTO = gradeObservable.body;
-                this.grade = gradeDTO.gradeName;
-                this.passed = gradeDTO.isPassingGrade;
-                this.isBonus = gradeDTO.gradeType === GradeType.BONUS;
-            }
-        });
+    setExamGrade(studentExamWithGrade: StudentExamWithGradeDTO) {
+        if (studentExamWithGrade?.studentResult?.overallGrade != undefined) {
```

Comment:
```suggestion
        if (studentExamWithGrade?.studentResult?.overallGrade !== undefined) {
```
Can the `!== undefined` be removed entirely, or would that produce wrong results for defined but empty strings being interpreted as `false`?

Reply:
`!==` would cause a `TypeError` for `null` values. It shouldn't be `null` anyway but I think for this case coding a little more defensively with `!=` would be more beneficial than a fail-fast, fail-hard approach when a `null` value is ever given. The empty string should also mean that the `overallGrade` is set on the server side so I would prefer keeping it this way since it also conveys the intent of the checked condition better in my opinion.